contracts/proxy.vy

```diff
@@ -0,0 +1,99 @@
+
+@public
+@constant
+def fee() -> decimal:
+    return convert(self.feeInt, decimal) / convert(FEE_MULTIPLIER, decimal)
+    
+# Note that due to rounding, the fees could be slightly higher for the tokens with smaller decimal precision.
+@public
+@constant
+def tokenExchangeRateAfterFees(input_token_address: address, output_token_address: address) -> uint256:
+    stablecoinswap_rate: uint256 = Stablecoinswap(self.stablecoinswapAddress).tokenExchangeRateAfterFees(input_token_address, output_token_address)
+    multiplier_after_fee: uint256 = FEE_MULTIPLIER - self.feeInt
+    exchange_rate: uint256 = stablecoinswap_rate * multiplier_after_fee / FEE_MULTIPLIER
+    return exchange_rate
+
+@public
+@constant
+def tokenOutputAmountAfterFees(input_token_amount: uint256, input_token_address: address, output_token_address: address) -> uint256:
+    # we can't just multiply all fees here because we round fee up
+    multiplier_after_fee: uint256 = FEE_MULTIPLIER - self.feeInt
+    input_amount_after_fee: uint256 = input_token_amount * multiplier_after_fee / FEE_MULTIPLIER
+    stablecoinswap_rate: uint256 = Stablecoinswap(self.stablecoinswapAddress).tokenExchangeRateAfterFees(input_token_address, output_token_address)
+
+    output_token_amount: uint256 = input_amount_after_fee * stablecoinswap_rate / EXCHANGE_RATE_MULTIPLIER
+    return output_token_amount
+
+# Trade one erc20 token for another
+@public
+@nonreentrant('lock')
+def swapTokens(input_token: address, output_token: address, erc20_input_amount: uint256, erc20_min_output_amount: uint256, deadline: timestamp) -> uint256:
+    transfer_from_user_result: bool = ERC20(input_token).transferFrom(msg.sender, self, erc20_input_amount)
+    assert transfer_from_user_result
+
+    multiplier_after_fee: uint256 = FEE_MULTIPLIER - self.feeInt
+    input_amount_after_fee: uint256 = erc20_input_amount * multiplier_after_fee / FEE_MULTIPLIER
+    fee_amount: uint256 = erc20_input_amount - input_amount_after_fee
+
+    # set allowance if necessary
+    stablecoinswap_allowance: uint256 = ERC20(input_token).allowance(self, self.stablecoinswapAddress)
+    if (stablecoinswap_allowance < input_amount_after_fee):
+        approval_result: bool = ERC20(input_token).approve(self.stablecoinswapAddress, 10**32)
+        assert approval_result
+
+    erc20_output_amount: uint256 = Stablecoinswap(self.stablecoinswapAddress).swapTokens(input_token, output_token, input_amount_after_fee, erc20_min_output_amount, deadline)
+    transfer_result: bool = ERC20(output_token).transfer(msg.sender, erc20_output_amount)
+    assert transfer_result
+    log.Trade(input_token, output_token, erc20_input_amount)
+    return erc20_output_amount
+
+@public
+def transferOwnership(new_owner: address) -> bool:
+    assert new_owner != ZERO_ADDRESS
+    assert msg.sender == self.owner
+    self.owner = new_owner
+    log.OwnershipTransferred(self.owner, new_owner)
+    return True
+
+@public
+def updateFee(value: decimal) -> bool:
```

Comment:
Can we switch this from decimal to uint256? It's very difficult to call this function accurately with the decimal argument from any of the currently available UIs (MEW, MyCrypto, etc)